File: src/snippets/bot-protection/reference/deno/DenyingBots.ts
import arcjet, { detectBot } from "@arcjet/deno";

const aj = arcjet({
  key: Deno.env.get("ARCJET_KEY")!, // Get your site key from https://app.arcjet.com
  rules: [
    detectBot({
      mode: "LIVE",
      // configured with a list of bots to deny from
      // https://arcjet.com/bot-list - all other detected bots will be allowed
      deny: [
        "CATEGORY:AI", // denies all detected AI and LLM scrapers
        "CURL", // denies the default user-agent of the `curl` tool
      ],
    }),
  ],
});

Deno.serve(
  { port: 3000 },
  aj.handler(async (req) => {
    const decision = await aj.protect(req);

    if (decision.isDenied()) {
      return new Response("Forbidden", { status: 403 });
    }

    return new Response("Hello world");
  }),
);

File: src/snippets/bot-protection/reference/python-fastapi/DenyingBots.py
import os

from arcjet import BotCategory, Mode, arcjet, detect_bot

aj = arcjet(
    key=os.environ["ARCJET_KEY"],
    rules=[
        detect_bot(
            mode=Mode.LIVE,
            # configured with a list of bots to deny from
            # https://arcjet.com/bot-list — all other detected bots will be allowed
            deny=[
                BotCategory.AI,  # denies all detected AI and LLM scrapers
                "CURL",  # denies the default user-agent of the `curl` tool
            ],
        ),
    ],
)
